### PR TITLE
DOC: Fix code that generates the table in the 'Casting Rules' section.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
The old code was Python 2 only.

### CODE REVIEW

File: doc/source/reference/ufuncs.rst [hunk 1]
```diff
@@ -228,45 +228,43 @@ can generate this table for your system with the code given in the Figure.
 
 .. admonition:: Figure
 
-    Code segment showing the "can cast safely" table for a 32-bit system.
+    Code segment showing the "can cast safely" table for a 64-bit system.
```
Comment (seberg) [inline doc/source/reference/ufuncs.rst]: If we want to be precise its a 64-bit system that is not windows :).

Comment (WarrenWeckesser) [reply doc/source/reference/ufuncs.rst]: OK, fixed (forced push).

**Comment by seberg** [comment]
Seems like a small enough fix, I am good if you just put it in when you feel you are done @WarrenWeckesser. The unicode characters are a nice touch. Hmmm, we have a script flying around somewhere that prints similar tables, but I am too stupid to find it right now.

File: doc/source/reference/ufuncs.rst (hunk 1 above)
Comment (seberg) [reply doc/source/reference/ufuncs.rst]: Maybe: `the output may depend on the system and for example differs on Windows?`, just thinking, I do not actually care too much :).

Comment (WarrenWeckesser) [reply doc/source/reference/ufuncs.rst]: That sounds better.

Comment (WarrenWeckesser) [reply doc/source/reference/ufuncs.rst]: Latest update is even more verbose. :)

File: doc/source/reference/ufuncs.rst [hunk 2]
```diff
@@ -228,45 +228,45 @@ can generate this table for your system with the code given in the Figure.
...
-    ...     for char in ntypes: print char,
-    ...     print
+    ...     print('X ' + ' '.join(ntypes))
     ...     for row in ntypes:
-    ...         print row,
+    ...         print(row, end='')
     ...         for col in ntypes:
-    ...             print int(np.can_cast(row, col)),
-    ...         print
-    >>> print_table(np.typecodes['All'])
```
Comment (seberg) [inline doc/source/reference/ufuncs.rst]: Sorry, did this line vanish completely? I had missed that.

Comment (WarrenWeckesser) [reply doc/source/reference/ufuncs.rst]: Oops, fixed.  I'll stop force-pushing until everything looks right.

Comment (seberg) [reply doc/source/reference/ufuncs.rst]: I honestly do not care for this small change. Lets juts get it done :).

**Comment by WarrenWeckesser** [comment]
Hmmm... looks like shippable took off early for the weekend.

**Comment by seberg** [comment]
OK, nvm shippable. Let me put this in, the snippet runs fine for me.

**Comment by WarrenWeckesser** [comment]
@seberg wrote
> we have a script flying around somewhere that prints similar tables,

Just happened to find it: [print_coercion_tables.py](https://github.com/numpy/numpy/blob/master/numpy/testing/print_coercion_tables.py)

**Comment by rgommers** [comment]
thanks! I've seen the output of that messing up failed tests recently, but I couldn't locate the source of it

**Comment by WarrenWeckesser** [comment]
Argh, apparently those Unicode checkmarks aren't drawn with the correct fixed-width when rendered on the web site (at least not with Firefox): https://numpy.org/devdocs/reference/ufuncs.html#casting-rules

If anyone sees a way to get the checkmarks to render correctly, let me know.  Otherwise, I'll change the checkmark to, say, the letter `Y` for *yes*.

**Comment by seberg** [comment]
Hmmm, that is annoying, it works fine for me. My guess is that your font is missing the character, so that it switches over to another one for it? But if that is not just an outlier, we need to replace.

**Comment by WarrenWeckesser** [comment]
As far as I can tell (a few minutes of searching makes me an expert, right?), there isn't such a thing as a monospace font for all Unicode.  Further investigation doesn't seem worthwhile at the moment.  I submitted a pull request to change the checkmark to `Y`: https://github.com/numpy/numpy/pull/14595

**Pull request merged**